core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/CoarseMesosSchedulerBackend.scala

```diff
@@ -204,35 +209,43 @@ private[spark] class CoarseMesosSchedulerBackend(
 
       for (offer <- offers) {
         val slaveId = offer.getSlaveId.toString
-        val mem = getResource(offer.getResourcesList, "mem")
-        val cpus = getResource(offer.getResourcesList, "cpus").toInt
-        if (totalCoresAcquired < maxCores &&
-            mem >= MemoryUtils.calculateTotalMemory(sc) &&
-            cpus >= 1 &&
+        var totalMem = getResource(offer.getResourcesList, "mem")
+        var totalCpus = getResource(offer.getResourcesList, "cpus").toInt
```

Comment:
since he changed the value of "offerCores" when some of the cores are allocated....a better name might be “leftCoresInOffer” ?

same for the memory